Add configuration docs for API restrictions (#24968)
cilium-team/cilium: Documentation/configuration/api-restrictions-table.rst

.. <!-- This file was autogenerated via api-flaggen, do not edit manually-->

Cilium Agent API
================

The following API flags are compatible with the ``cilium-agent`` flag
``enable-cilium-api-server-access``.

===================== ====================
Flag Name             Description
===================== ====================

> DeleteEndpointID      Deletes the endpoint specified by the ID. Deletion is
>                       imminent and atomic, if the deletion request is valid and
>                       the endpoint exists, deletion will occur even if errors are
>                       encountered in the process. If errors have been encountered,
>                       the code 202 will be returned, otherwise 200 on success. All
>                       resources associated with the endpoint will be freed and the
>                       workload represented by the endpoint will be disconnected.It
>                       will no longer be able to initiate or receive communications
>                       of any sort.

Reviewer: Edits for clarity, use present tense
```suggestion
DeleteEndpointID      Deletes the endpoint specified by the ID. Deletion is
                      imminent and atomic: if the deletion request is valid and
                      the endpoint exists, deletion occurs even if errors are
                      encountered in the process. If errors are encountered,
                      HTTP code 202 is returned, otherwise 200 returns on success. Upon deletion,
                      all resources associated with the endpoint are freed and the
                      workload represented by the endpoint is disconnected. The endpoint
                      is no longer able to initiate or receive communications
                      of any sort.
```

> DeleteFqdnCache       Deletes matching DNS lookups from the cache, optionally
>                       restricted by DNS name. The removed IP data will no longer
>                       be used in generated policies.

Reviewer: clarity, present tense
```suggestion
DeleteFqdnCache       Deletes the matching DNS lookups from the cache, optionally
                      restricted by DNS name. The removed IP data is no longer
                      used in generated policies.
```

DeleteIPAMIP          -
DeletePolicy          -
DeletePrefilter       -
DeleteRecorderID      -
DeleteServiceID       -

> GetBGPPeers           Retrieves current operational state of BGP peers created by
>                       Cilium BGP virtual router. This includes session state,
>                       uptime, information per address family, etc.

Reviewer: clarity, no Latin abbreviations
```suggestion
GetBGPPeers           Retrieves the current operational state of BGP peers created by
                      the Cilium BGP virtual router. This includes session state,
                      uptime, information per address family, and others.
```

GetCgroupDumpMetadata -
GetClusterNodes       -
GetConfig             Returns the configuration of the Cilium daemon.
GetDebuginfo          -
GetEndpoint           Retrieves a list of endpoints that have metadata matching
                      the provided parameters, or all endpoints if no parameters
                      provided.

> GetEndpointID         Returns endpoint information

Reviewer: Use periods consistently
```suggestion
GetEndpointID         Returns endpoint information.
```

GetEndpointIDConfig   Retrieves the configuration of the specified endpoint.
GetEndpointIDHealthz  -
GetEndpointIDLabels   -
GetEndpointIDLog      -
GetFqdnCache          Retrieves the list of DNS lookups intercepted from
                      endpoints, optionally filtered by DNS name, CIDR IP range or
                      source.
GetFqdnCacheID        Retrieves the list of DNS lookups intercepted from the
                      specific endpoint, optionally filtered by endpoint id, DNS
                      name, CIDR IP range or source.
GetFqdnNames          Retrieves the list of DNS-related fields (names to poll,
                      selectors and their corresponding regexes).
GetHealthz            Returns health and status information of the Cilium daemon
                      and related components such as the local container runtime,
                      connected datastore, Kubernetes integration and Hubble.

> GetIP                 Retrieves a list of IPs with known associated information
>                       such as their identities, host addresses, Kubernetes pod
>                       names, etc. The list can optionally filtered by a CIDR IP
>                       range.

Reviewer: No Latin abbreviations
```suggestion
GetIP                 Retrieves a list of IPs with known associated information
                      such as their identities, host addresses, Kubernetes pod
                      names, and others. The list can optionally filtered by a CIDR IP
                      range.
```

> GetIdentity           Retrieves a list of identities that have metadata matching
>                       the provided parameters, or all identities if no parameters
>                       are provided.

Reviewer: Use active voice
```suggestion
GetIdentity           Retrieves a list of identities that have metadata matching
                      the provided parameters, or all identities if you provide no parameters.
```

GetIdentityEndpoints  -
GetIdentityID         -
GetLRP                -
GetMap                -
GetMapName            -
GetMapNameEvents      -
GetMetrics            -
GetNodeIds            Retrieves a list of node IDs allocated by the agent and
                      their associated node IP addresses.
GetPolicy             Returns the entire policy tree with all children.
GetPolicySelectors    -
GetPrefilter          -
GetRecorder           -
GetRecorderID         -
GetRecorderMasks      -
GetService            -
GetServiceID          -

> PatchConfig           Updates the daemon configuration by applying the provided
>                       ConfigurationMap and regenerates & recompiles all required
>                       datapath components.

Reviewer: abbreviation nit
```suggestion
PatchConfig           Updates the daemon configuration by applying the provided
                      ConfigurationMap and regenerates and recompiles all required
                      datapath components.
```

> PatchEndpointID       Applies the endpoint change request to an existing endpoint

Reviewer: Use periods consistently
```suggestion
PatchEndpointID       Applies the endpoint change request to an existing endpoint.
```

> PatchEndpointIDConfig Update the configuration of an existing endpoint and
>                       regenerates & recompiles the corresponding programs
>                       automatically.

Reviewer: clarity, nit
```suggestion
PatchEndpointIDConfig Updates the configuration of an existing endpoint and
                      automatically regenerates and recompiles the corresponding programs.
```

PatchEndpointIDLabels Sets labels associated with an endpoint. These can be user
                      provided or derived from the orchestration system.
PatchPrefilter        -
PostIPAM              -
PostIPAMIP            -

> PutEndpointID         Creates a new endpoint

Reviewer: Use periods consistently
```suggestion
PutEndpointID         Creates a new endpoint.
```

===================== ====================

Cilium Agent Clusterwide Health API
===================================

The following API flags are compatible with the ``cilium-agent`` flag
``enable-cilium-health-api-server-access``.

===================== ====================
Flag Name             Description
===================== ====================
GetHealthz            Returns health and status information of the local node
                      including load and uptime, as well as the status of related
                      components including the Cilium daemon.
GetStatus             Returns the connectivity status to all other cilium-health
                      instances using interval-based probing.
PutStatusProbe        Runs a synchronous probe to all other cilium-health
                      instances and returns the connectivity status.
===================== ====================

Cilium Operator API
===================

The following API flags are compatible with the ``cilium-operator`` flag
``enable-cilium-operator-server-access``.

===================== ====================
Flag Name             Description
===================== ====================

> GetHealthz            This path will return the status of cilium operator

Reviewer: Use present tense, clarity
```suggestion
GetHealthz            This path returns the status of the cilium operator
```
